src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SeparateSidewalkTagCheck.java

```diff
@@ -0,0 +1,341 @@
+     */
+    private boolean isRightOf(final PolyLine line, final Location location)
+    {
+        final PolyLine locationLine = new PolyLine(location);
+        final Segment closest = line.segments().stream()
+                .min(Comparator.comparingDouble(s -> s.shortestDistanceTo(locationLine).asMeters()))
+                .orElse(null);
+        if (closest != null)
+        {
+            final PolyLine testLine = new PolyLine(closest.first(), closest.last(), location);
+            final Angle difference = testLine.headingDifference().orElse(null);
+            return difference != null && difference.asDegrees() > 0;
+        }
+        return false;
+    }
+
+    private boolean isSidewalkTaggingMismatch(final String sidewalkTagValue,
+            final int rightSidewalkCount, final int leftSidewalkCount)
+    {
+        switch (sidewalkTagValue)
+        {
+            // sidewalk=right but separate sidewalks on left or both sides are detected
+            case "right":
+                return leftSidewalkCount > 0;
+            // sidewalk=left but separate sidewalks on right or both sides are detected
+            case "left":
+                return rightSidewalkCount > 0;
+            // sidewalk=both but separate sidewalk on left or ride side is detected.
+            case "both":
+                return rightSidewalkCount < 1 || leftSidewalkCount < 1;
+            default:
+                throw new CoreException("Unknown sidewalk tag value {}", sidewalkTagValue);
+        }
+    }
+
+    /**
+     * Helper function for filtering {@link Edge}s. This is detecting that {@link Edge} is
+     * separately mapped sidewalk https://wiki.openstreetmap.org/wiki/Sidewalks or designated
+     * pedestrian road https://wiki.openstreetmap.org/wiki/Key:foot.
+     *
+     * @param edge
+     *            Edge to examine
+     * @return true if {@link Edge} is passed validation.
+     */
+    @SuppressWarnings("squid:S3655")
+    private boolean validFootwayFilter(final Edge edge)
+    {
+        return edge.isMainEdge()
+                && (edge.getTag(HighwayTag.KEY).isPresent()
+                        && "footway".equals(edge.getTag(HighwayTag.KEY).get())
+                        && edge.getTag(FootwayTag.KEY).isPresent()
+                        && "sidewalk".equals(edge.getTag(FootwayTag.KEY).get()))
+                || (edge.getTag(FootTag.KEY).isPresent()
+                        && edge.getTag(FootTag.KEY).get().matches("yes|designated"));
+    }
+
+    /**
+     * Helper function for filtering {@link Edge}s. This is to detect {@link Edge} sidewalk tag
+     * existence with other
+     *
+     * @param edge
+     *            Edge to examine
+     * @return {@code true} if {@link Edge} sidewalk tag meet the criteria.
+     */
+    @SuppressWarnings("squid:S3655")
+    private boolean validSidewalkFilter(final Edge edge)
+    {
+        return edge.getTag(SidewalkTag.KEY).isPresent() && !this.isDualCarriageWay(edge)
```

Comment:
@vladlemberg, your testInvalidHighwayLeftSidewalkRightAlternativeMapping is failing because `edge.getTag(SidewalkTag.KEY).isPresent()` is false for all features in its test atlas.